site/gatsby-site/src/components/cite/RemoveDuplicateModal.js

```diff
@@ -0,0 +1,194 @@
+                            .concat(true_incident_number),
+                        },
+                        reports: {
+                          link: classification.reports.map((report) => report.report_number),
+                        },
+                      },
+                    },
+                  });
+                }
+              } catch (e) {
+                addToast({
+                  message: `Could not transfer classifications to incident ${true_incident_number}.`,
+                  severity: SEVERITY.danger,
+                  error: e,
+                });
+              }
+              try {
+                for (const subscription of subscriptionsData.subscriptions) {
+                  await updateSubscription({
+                    variables: {
+                      query: { _id: subscription._id },
+                      subscription: {
+                        userId: { link: subscription.userId.userId },
+                        incident_id: { link: true_incident_number },
+                        type: subscription.type,
+                      },
+                    },
+                  });
+                }
+              } catch (e) {
+                addToast({
+                  message: `Could not transfer subscription to incident ${true_incident_number}.`,
+                  severity: SEVERITY.danger,
+                  error: e,
+                });
+              }
+
+              setSubmitting(false);
+              onClose();
+              addToast({
+                severity: SEVERITY.success,
+                message: [
+                  `Incident ${incident.incident_id} marked`,
+                  `as duplicate of ${values.duplicateIncidentId[0]}.`,
+                  `Its page will updated within 24 hours.`,
+                ].join(' '),
+              });
+              setTimeout(() => (window.location.pathname = '/'), 5000);
+            };
+
+            return (
+              <Form>
+                <label htmlFor="duplicateIncidentId">
+                  <Trans>Transfer reports to incident</Trans>{' '}
+                </label>
+                <IncidentsField
+                  multiple={false}
+                  id="duplicateIncidentId"
+                  name="duplicateIncidentId"
+                  className="mb-4 mt-1"
+                />
+                <Button
+                  color="failure"
+                  disabled={submitBlocked}
+                  onClick={submit}
+                  data-cy="confirm-remove-duplicate"
+                >
+                  <Trans>Remove Duplicate</Trans>
+                </Button>
+              </Form>
+            );
+          }}
+        </Formik>
+      </Modal.Body>
+    </Modal>
+  );
+}
+
+const removeTypename = (obj) => {
+  const replaced = JSON.stringify(obj).replace(/"__typename":"[A-Za-z]*",/g, '');
+
+  return JSON.parse(replaced);
+};
```

Comment:
Please try to use this [removeTypenameFromVariables](https://www.apollographql.com/docs/react/api/link/apollo-link-remove-typename/)  Apollo function

Reply:
I naïvely thought that a function called `removeTypenameFromVariables` would take variables as input and remove the typenames from them. But actually it's a builder for an `ApolloLink` object that you're supposed to pass to the `ApolloClient` constructor. However, if I understand correctly, we have one client that we use everywhere through hooks. So this requires either instantiating a new client just for this, which would be annoying, or removing the typenames everywhere, which I think is ok but I would want to check with everyone. The change is made [here](https://github.com/responsible-ai-collaborative/aiid/pull/2130/files#diff-669c71f154e5d21df106aeb41c82230771bf55c23c69c1a84e175fff5c5fa8eb).

Reply:
That's great. We don't use those `_typenames` fields anywhere, so if that approach using ApolloLink works, we can clean up the code where we delete them manually.